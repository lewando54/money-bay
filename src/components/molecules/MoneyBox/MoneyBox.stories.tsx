/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import MoneyBox from './MoneyBox'

const meta = {
    component: MoneyBox,
    argTypes: {
        name: { control: 'text' },
        goalAmount: { control: 'number' },
        currentAmount: { control: 'number' },
        currency: { control: 'object' },
        onTopUpClick: { action: 'topped-up' },
        onWithdrawalClick: { action: 'withdrawal' }
    }
} satisfies Meta<typeof MoneyBox>

export default meta
type Story = StoryObj<typeof MoneyBox>

export const Default: Story = {
    args: {
        goalAmount: 1200,
        currentAmount: 650.27,
        name: 'New iPhone Pro Max',
        currency: {iso: 'usd', symbol: '$', left: true},
    }
}
