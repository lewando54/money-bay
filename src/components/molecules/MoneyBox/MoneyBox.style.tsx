import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    container: {
        gap: 10,
        padding: 21
    },
    titleAndGoalContainer: {
        flexDirection: 'row',
        gap: 7,
        alignItems: 'center'
    },
    goalContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        marginLeft: 'auto'
    },
    floatCurrencyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        paddingBottom: 2
    },
    floatCurrentCurrencyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        paddingBottom: 1
    },
    currentMoneyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        justifyContent: 'center',
        marginTop: 5
    },
    buttonsContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        gap: 8
    },
    progressBar: {
        borderRadius: 10,
        height: 5
    },
    moneyText: {
        marginLeft: 'auto'
    }
})
