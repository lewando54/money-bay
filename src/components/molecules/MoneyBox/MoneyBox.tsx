import React from 'react'
import { View, Text } from 'react-native'
import { TCurrency } from '../../../utils/currency'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import moneyBoxStyle from './MoneyBox.style'
import { ProgressBar } from 'react-native-paper'
import globalThemeStyle, { GOOD_GREEN, MAIN_DARK, PLAIN_BLUE } from '../../../styling/GlobalTheme.style'
import PiggyBank from './assets/piggy-bank.svg'
import Button from '../../atoms/Button/Button'
import MoneyText from '../../atoms/MoneyText/MoneyText'

interface IMoneyBoxProps {
    goalAmount?: number
    currentAmount?: number
    name?: string
    currency?: TCurrency
    testID?: string
    onTopUpClick?: () => void
    onWithdrawalClick?: () => void
}

export default function MoneyBox({
    goalAmount=1200,
    currentAmount=650.27,
    name='New iPhone Pro Max',
    currency={iso: 'usd', symbol: '$', left: true},
    testID,
    onTopUpClick,
    onWithdrawalClick
}: IMoneyBoxProps){
    const verifiedGoalAmount = 
        goalAmount <= 0 ? 
            1 : 
            goalAmount
    const verifiedCurrentAmount = 
        currentAmount > goalAmount ? 
            goalAmount : 
            currentAmount <= 0 ? 
                1 : 
                currentAmount

    return (
        <UniversalContainer testID={testID} style={moneyBoxStyle.container} castShadow={true}>
            <View style={moneyBoxStyle.titleAndGoalContainer}>
                <PiggyBank />
                <Text style={[globalThemeStyle.text_Regular, {color: MAIN_DARK}]}>{name}</Text>
                <MoneyText amount={verifiedGoalAmount} currency={currency} containerStyle={{marginLeft: 'auto'}}/>
            </View>
            <View>
                <ProgressBar progress={(verifiedCurrentAmount/verifiedGoalAmount)} style={{borderRadius: 10, height: 5}} color={GOOD_GREEN}/>
                
                <View style={moneyBoxStyle.currentMoneyContainer}>
                    <MoneyText amount={verifiedCurrentAmount} currency={currency} fontSize={12} isCurrencyBold={false}/>
                </View>
            </View>
            <View style={moneyBoxStyle.buttonsContainer}>
                <Button width='50%' color={PLAIN_BLUE} size='small' onClick={onTopUpClick}>Top - Up</Button>
                <Button width='50%' color={GOOD_GREEN} size='small' onClick={onWithdrawalClick}>Withdrawal</Button>
            </View>
        </UniversalContainer>
    )
}
