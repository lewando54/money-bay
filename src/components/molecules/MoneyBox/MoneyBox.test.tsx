import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import MoneyBox from './MoneyBox'
describe('MoneyBox', () => {
    jest.useFakeTimers()
    it('should render correctly', () => {
        const tree = renderer.create(<MoneyBox />)
        expect(tree).toMatchSnapshot()
    })
})
