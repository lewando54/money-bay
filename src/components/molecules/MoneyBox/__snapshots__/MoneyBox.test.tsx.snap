// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyBox should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#161966",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "gap": 10,
        "padding": 21,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 7,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
          },
        ]
      }
    >
      New iPhone Pro Max
    </Text>
    <View
      style={
        {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "marginLeft": "auto",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 20,
            },
          ]
        }
      >
        1200
        .
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "paddingBottom": 2,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
              },
            ]
          }
        >
          00
        </Text>
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishBold",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
                "marginLeft": 4,
              },
            ]
          }
        >
          USD
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityRole="progressbar"
      accessibilityState={
        {
          "busy": true,
        }
      }
      accessibilityValue={
        {
          "max": 100,
          "min": 0,
          "now": 54.189166666666665,
        }
      }
      accessible={true}
      onLayout={[Function]}
      style={false}
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderRadius": 10,
            "height": 5,
            "opacity": 0,
            "overflow": "hidden",
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "fontSize": 12,
            },
          ]
        }
      >
        650
        .
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "paddingBottom": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 12,
              },
              {
                "fontSize": 8,
              },
            ]
          }
        >
          27
        </Text>
        <Text
          style={
            [
              {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 12,
              },
              {
                "fontSize": 8,
                "marginLeft": 3,
              },
            ]
          }
        >
          USD
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4283804910,
              4283804910,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            {
              "padding": 8,
            },
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              false,
              false,
              [
                {
                  "fontSize": 12,
                },
                {
                  "color": "white",
                  "fontFamily": "MulishSemiBold",
                },
              ],
            ]
          }
        >
          Top - Up
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4282299024,
              4282299024,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            {
              "padding": 8,
            },
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              false,
              false,
              [
                {
                  "fontSize": 12,
                },
                {
                  "color": "white",
                  "fontFamily": "MulishSemiBold",
                },
              ],
            ]
          }
        >
          Withdrawal
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
